data/updater.py

```diff
@@ -0,0 +1,357 @@
+sZ094LwkO8GRySB+LKm6KQtJ+WOnsvs3X8v8fSA6GwJjYdtKqNUzPBLpw8RrIH9l
+eaT2pe9Ta48GqEwrU8wxwKyRBIfJJP/zq5n1rKcOBpvLZDVcyrVw+pIGa0zfmr/c
+qWYG7znx2Xq3i22d36xPkfkZEyVnQcCJJ28hkAfXRYpp+gMnL0Zt4u3GgzSARSBS
+VrcMyNlaft/aSOkojyjh3+2zF1PCfW1Nw9Sx50gdN3FfF0yEWjUoA1R/NW9CQZVG
+4qh/n2k508PYfZRuJ74T2jABFJIztv2pmq3VpSA7hkHGl3nXrdqpsw3V9bkFqZa/
+ihhY7IpGwUWx4pDHh1gKhjJ0qPUVK5sOx3GZfEvMCCiH9XPk70fn3nuYupRr9WNr
+HJwUSeLMhRvi4jTT+z5QLdYloFRZmDRwNg63csGZRkly9vjrAiMVHMpcJI0eCei/
+XgeKSxoiAmzNuc2J47SF2z7WIsDwHhwRj6tj4dOW3Ye0WIkcTIvHd7UTVX02v+oB
+d5YhABEBAAGJAiUEGAEKAA8FAlfEAKACGwwFCQHhM4AACgkQw+zcBCBPnSn25BAA
+xU7NKi6BokqnloYncvL74fuUpam3LJBwsOkFehuO2D+X9A2blIpbpXtUoWXwRc9V
+Jf7nL/yhMKcOB9m1MHVPtxtN5JzV9p8k2BT04/X6fa09umsJ3hwg/zhXrkGFrMVE
+hfAk6q8a0Y5oJKUOoJhzxqD9ItibxHPkqb+R/GSMfrDIRE0ecfIltDLIRQMlBF1b
+z5WN5Dwv8cikeQrsK6DNvbUUuHAbG8RZYG9QxFdkbehp46bCA8CfINENBzIskckx
+xwlTtVxcDD0Irql7EuIM1bpWdFxBZPlmcDyLrZgCYgSqPOe2mOK9V37jM3fsTqR9
+C9fmr3DDmAm8XrUBL9ORwMTFa6LIUpUoSSzG258h9qPiBySj7b51QbrJ9WXiISsQ
+5X/V8u6Qbp9PJnEUXXfIE9YVspF3+Zrfn0XqqjEND+bWi52cgDvwROSFf4KQnAto
+FdlxHtxlcZgJEFeHLyb370XI4rcNs7zMoD3B06Eyay5oY98w9JFYwrm6bVvNzis7
+CrK2K46QzSUZm1BHTOi0AlnqpYtLaJnMSQCxTmjyRBaJM5DGXs/86y/OYdBcpiYG
+aV0q9EcvDLQnSpa0vlsVM0AfMY+To1RTCMv+TGKMZHHWeV2yUABWK52raBpQHtjs
+7SonBqg04+2E4w1WmZEx1u9QyDXmlaLxnR+YIqilM7g=
+=RbWA
+-----END PGP PUBLIC KEY BLOCK-----
+"""
+
+
+class RequestFailed(Exception):
+    pass
+
+def get_request(url, follow_redirects=True):
+    cmd = ["curl", "-q"]
+    if follow_redirects is True:
+        cmd.append("-L")
+    cmd.append(url)
+
+    tmp_file = tempfile.TemporaryFile()
+
+    try:
+        check_call(cmd, stdout=tmp_file)
+    except CalledProcessError:
+        raise RequestFailed
+
+    tmp_file.seek(0)
+
+    return tmp_file.read()
+
+def get_current_version():
+    if not os.path.exists(CURRENT_VERSION_PATH):
+        return 0
+    with open(CURRENT_VERSION_PATH) as in_file:
+        version = in_file.read()
+    return int(version)
+
+def get_latest_version():
+    version = get_request(UPDATE_BASE_URL + "latest/version")
+    return int(version)
+
+class InvalidSignature(Exception):
+    pass
+
+class InvalidPublicKey(Exception):
+    pass
+
+
+def verify_file(signature_path, file_path, signer_pk_path):
+    tmp_dir = tempfile.mkdtemp()
+    tmp_key = os.path.join(tmp_dir, "signing-key.gpg")
+
+    try:
+        try:
+            check_call(["gpg", "--batch", "--yes", "-o", tmp_key,
+                        "--dearmor", signer_pk_path])
+        except CalledProcessError:
+            raise InvalidPublicKey
+
+        try:
+            output = check_output(["gpg", "--batch", "--status-fd", "1",
+                                   "--no-default-keyring", "--keyring",
+                                   tmp_key, "--trust-model", "always",
+                                   "--verify", signature_path, file_path])
+        except CalledProcessError:
+            raise InvalidSignature
```

Comment:
The basically convert an ascii armored key into a binary key that can be used as a keyring so that we can verify the file by pointing to the binary key as a keyring.